builtin/providers/aws/resource_aws_route53_record.go

```diff
@@ -732,7 +733,7 @@ func expandRecordName(name, zone string) string {
 func resourceAwsRoute53AliasRecordHash(v interface{}) int {
 	var buf bytes.Buffer
 	m := v.(map[string]interface{})
-	buf.WriteString(fmt.Sprintf("%s-", m["name"].(string)))
+	buf.WriteString(fmt.Sprintf("%s-", normalizeAwsAliasName(m["name"].(string))))
 	buf.WriteString(fmt.Sprintf("%s-", m["zone_id"].(string)))
```

Comment:
Sorry for the late comment here @stack72... but when I've seen this class of error so far I've seen `zone_id` change too, as if Route53 uses a different zone for the `dualstack.` form of the hostname than for the regular form. You can see this e.g. in the example in the summary of #9628.

Do you think we need to do something additional here to catch that case, or is this fine because the non-prefixed hostname happens to resolve in the other zone too?